src/core/Song.cpp

```diff
@@ -813,6 +813,83 @@ void Song::addPatternTrack()
 	Engine::patternStore()->setCurrentPattern(dynamic_cast<PatternTrack*>(t)->patternIndex());
 }
 
+/* @brief Converts all PatternTracks to regular tracks on the Song Editor
+ * @param bool singlePattern Indicates whether we are converting a single
+ * PatternTrack or all of the Song PatternTracks.
+ * @param int pattern If single Pattern is true, indicates the index of the
+ * PatternTrack being converted.
+ */
+void Song::convertPatterntoSE(bool singlePattern /* = false*/, int pattern /* = 0*/)
+{
+	// Get all tracks from the Pattern editor
+	PatternStore* patternStore = Engine::patternStore();
+	TrackList tl = patternStore->tracks();
+
+	// For each track, go through all clips and copy them in every place
+	// the corresponding PatternTrack has PatternClips in.
+	for
+	(
+		TrackList::iterator it = tl.begin();
+		it != tl.end();
+		++it
+	)
+	{
+		// Create a clone of the track and add it to the song editor
+		Track* clonedTrack = nullptr;
+		if ((*it)->type() == Track::InstrumentTrack || (*it)->type() == Track::SampleTrack)
+		{
+			// Create XML data of the track for the clone
+			QDomDocument doc;
+			QDomElement parent = doc.createElement("clone");
+			(*it)->saveState(doc, parent);
+			clonedTrack = Track::create(parent.firstChild().toElement(), this);
+			clonedTrack->deleteClips();
+		}
+		else { continue; }
+
+		// Convert either a single PatternTrack or all of them
+		auto numClips = (*it)->numOfClips();
+		int firstPattern = singlePattern ? pattern : 0;
+		int lastPattern = singlePattern ? pattern : numClips - 1;
+
+		// Now go through all the Clips and copy them in the appropriate place
+		// in the cloned track
+		for (int i = firstPattern; i <= lastPattern; ++i)
+		{
+			// Find the PatternTrack
+			PatternTrack* pt = PatternTrack::findPatternTrack(i);
+			if (!pt) { continue; }
+
+			// Get the Clip from the track inside the PatternTrack
+			Clip* src = (*it)->getClip(i);
+
+			// If Clip is a MidiClip and it's empty, skip it
+			MidiClip* mc = dynamic_cast<MidiClip *>(src);
+			if (mc && mc->empty()) { continue; }
+
+			// Go through the PatternTrack's Clips
+			Track::clipVector patternclips = pt->getClips();
```

Comment:
```suggestion
			const auto& patternclips = pt->getClips();
```
`getClips()` returns a const reference, so you'll want to store it as a const reference to avoid an unnecessary copy of the entire clips vector. This is due to the `QVector` to `std::vector` refactor which happened recently.